src/anyio/_backends/_asyncio.py

```diff
@@ -2711,7 +2711,7 @@ async def wait_socket_writable(cls, sock: socket.socket) -> None:
 
         loop = get_running_loop()
         event = write_events[sock] = asyncio.Event()
```

Comment:
I think write events now needs to be a RunVar of a dict of int to Event, rather than Any to Event. Otherwise someone could pass a socket and then later pass that socket's fd

Reply:
I concur – the key needs to be the `fileno()` of the socket.